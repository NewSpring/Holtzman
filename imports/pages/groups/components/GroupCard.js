import PropTypes from 'prop-types';
import truncate from "truncate";
import { withRouter } from "react-router";

import SideBySide from "../../../components/@primitives/UI/cards/SideBySideCard";

import Tag from "../../../components/@primitives/UI/tags";

export const GroupCardWithoutData = ({ group, router, onHover }) => {
  const theGroup = group || {};

  return (
    <button
      onMouseOver={onHover}
      id={theGroup.id}
      onClick={() => router.push(`/groups/${theGroup.id}`)}
      className="relative one-whole push-bottom@lap-and-up plain"
    >
      <SideBySide
        defaultImage={
          theGroup.photo ?
            theGroup.photo :
            "//s3.amazonaws.com/ns.assets/apollos/group-profile-placeholder.png"
        }
        left={["one-whole", "two-thirds@lap-and-up"]}
        right={["one-whole", "one-third@lap-and-up"]}
      >
        {/* Name */}
        <h4 className="plain text-dark-primary push-half-top push-top@anchored">
          {theGroup.name}
        </h4>

        {/* Schedule */}
        {theGroup.schedule && theGroup.schedule.description && (
          <h6 className="plain text-dark-tertiary">
            {theGroup.schedule.description}
          </h6>
        )}

        {/* Distance */}
        {theGroup.distance && (
          <h6 className="em text-dark-tertiary push-half-bottom">
            {theGroup.distance.toFixed(2)} miles away
          </h6>
        )}


        {/* Description */}
        <p className="plain text-dark-primary">{truncate(theGroup.description, 120)}</p>

        {/* Tags */}
        <div className="soft-half-top">
          {theGroup.tags && theGroup.tags.filter(x => x).map((tag, i) => (
            <Tag val={tag.value} key={i} />
          ))}

          {theGroup.type && theGroup.type !== "Interests" && <Tag val={theGroup.type} />}

          {theGroup.kidFriendly && <Tag val="kid friendly" />}

          {theGroup.demographic && <Tag val={group.demographic} />}

          {group.campus && group.campus.name && <Tag val={group.campus.name} urlKey="campus" />}

        </div>

      </SideBySide>
    </button>
  );
};

GroupCardWithoutData.propTypes = {
  group: PropTypes.object,
  router: PropTypes.object,
  onHover: PropTypes.func,
};

export default withRouter(({ group, router, onHover }) => (
  <GroupCardWithoutData group={group} router={router} onHover={onHover} />
));

