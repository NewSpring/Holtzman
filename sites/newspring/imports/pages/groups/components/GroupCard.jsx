import Truncate from "truncate";
import { withRouter } from "react-router"

import Loading from "apollos/dist/core/components/loading/index";
import SideBySide from "apollos/dist/core/components/cards/SideBySide";

import Tag from "../components/Tag";

export default withRouter(({ group, router, onHover }) => {
  if (!group) group = {};
  const leaders = group && group.members && group.members
    .filter(x => x.role.toLowerCase() === "leader");

  return (
    <button
      onMouseOver={onHover}
      id={group.id}
      onClick={() => router.push(`/groups/${group.id}`)}
      className="relative one-whole push-bottom@lap-and-up plain"
    >
      <SideBySide
        defaultImage={group.photo ? group.photo : "//s3.amazonaws.com/ns.assets/apollos/group-profile-placeholder.png" }
        left={["one-whole", "two-thirds@lap-and-up"]}
        right={["one-whole", "one-third@lap-and-up"]}
      >
        {/* Name */}
        <h4 className="plain text-dark-primary push-half-top push-top@anchored">
          {group.name}
        </h4>

        {/* Leaders */}
        <h5 className="plain text-dark-tertiary">
          {leaders && leaders.map((x, i) => {
            let string = `${x.person.nickName || x.person.firstName} ${x.person.lastName}`;
            if (leaders.length - 1 != i) string += ", ";
            return <span key={i}>{string}</span>
          })}</h5>

        {/* Distance */}
        {(() => {
          if (!group.distance) return null;
          return (
            <h6 className="em text-dark-tertiary push-half-bottom">
              {group.distance.toFixed(2)} miles away
            </h6>
          )
        })()}


        {/* Description */}
        <p className="plain text-dark-primary">{Truncate(group.description, 120)}</p>

        {/* Tags */}
        <div className="soft-half-top">
          {group.tags && group.tags.filter(x => x).map((tag, i) => (
            <Tag val={tag.value} key={i} />
          ))}
          {(() => {
            if (!group.type) return null;
            return <Tag val={group.type} />
          })()}
          {/*{(() => {
            if (!group.kidFriendly) return null;
            return <Tag value="childcare" />
          })()}*/}
          {(() => {
            if (!group.demographic) return null;
            return <Tag val={group.demographic} />
          })()}
        </div>

      </SideBySide>
    </button>
  )
})
