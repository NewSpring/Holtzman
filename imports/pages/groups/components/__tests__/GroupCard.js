import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { clone } from "ramda";
import { GroupCardWithoutData as GroupCard } from "../GroupCard";

const defaultProps = {
  group: {
    id: "1",
    photo: "http://test.com/test.jpg",
    name: "test group",
    schedule: {
      description: "test description",
    },
    distance: 2,
    tags: [
      { value: "one" },
      { value: "two" },
    ],
    type: "test type",
    kidFriendly: true,
    demographic: "test demographic",
    campus: {
      name: "Anderson",
    },
  },
  router: {
    push: jest.fn(),
  },
  onHover: jest.fn(),
};

const generateComponent = (additionalProps ={}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <GroupCard { ...newProps } />;
};

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders default photo", () => {
  const props = clone(defaultProps);
  props.group.photo = null;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders without schedule", () => {
  const props = clone(defaultProps);
  props.group.schedule = null;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders without schedule description", () => {
  const props = clone(defaultProps);
  props.group.schedule.description = null;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders without distance", () => {
  const props = clone(defaultProps);
  props.group.distance = null;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("doesn't render tag for type `Interests`", () => {
  const props = clone(defaultProps);
  props.group.type = "Interests";
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("doesn't render kid friendly tag if no", () => {
  const props = clone(defaultProps);
  props.group.kidFriendly = false;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("doesn't render demographic tag if no", () => {
  const props = clone(defaultProps);
  props.group.demographic = false;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("doesn't render campus tag if no campus", () => {
  const props = clone(defaultProps);
  props.group.campus = false;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("doesn't render campus tag if no campus name", () => {
  const props = clone(defaultProps);
  props.group.campus.name = false;
  const wrapper = shallow(generateComponent(props));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
