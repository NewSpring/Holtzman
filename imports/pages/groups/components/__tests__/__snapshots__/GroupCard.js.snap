exports[`test doesn't render campus tag if no campus 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
    </div>
  </Card>
</button>
`;

exports[`test doesn't render campus tag if no campus name 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
    </div>
  </Card>
</button>
`;

exports[`test doesn't render demographic tag if no 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test doesn't render kid friendly tag if no 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test doesn't render tag for type \`Interests\` 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test renders default photo 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="//s3.amazonaws.com/ns.assets/apollos/group-profile-placeholder.png"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test renders with props 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test renders without distance 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="plain text-dark-tertiary">
      test description
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test renders without schedule 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;

exports[`test renders without schedule description 1`] = `
<button
  className="relative one-whole push-bottom@lap-and-up plain"
  id="1"
  onClick={[Function]}
  onMouseOver={[Function]}>
  <Card
    defaultImage="http://test.com/test.jpg"
    left={
      Array [
        "one-whole",
        "two-thirds@lap-and-up",
      ]
    }
    right={
      Array [
        "one-whole",
        "one-third@lap-and-up",
      ]
    }>
    <h4
      className="plain text-dark-primary push-half-top push-top@anchored">
      test group
    </h4>
    <h6
      className="em text-dark-tertiary push-half-bottom">
      2.00
       miles away
    </h6>
    <p
      className="plain text-dark-primary" />
    <div
      className="soft-half-top">
      <withRouter(Connect(TagWithoutData))
        val="one" />
      <withRouter(Connect(TagWithoutData))
        val="two" />
      <withRouter(Connect(TagWithoutData))
        val="test type" />
      <withRouter(Connect(TagWithoutData))
        val="kid friendly" />
      <withRouter(Connect(TagWithoutData))
        val="test demographic" />
      <withRouter(Connect(TagWithoutData))
        urlKey="campuses"
        val="Anderson" />
    </div>
  </Card>
</button>
`;
